### PR TITLE
http2: Add integration tests for METADATA and RST_STREAM frame flood mitigation for upstream servers

### PR DESCRIPTION
**Pull request by yanavlasov**
Risk Level: Low, tests only
Testing: Integration Test
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

Part of #12281

Signed-off-by: Yan Avlasov <yavlasov@google.com>

### CODE REVIEW

File: test/integration/http2_flood_integration_test.cc [hunk 1]
```diff
@@ -274,6 +275,52 @@ void Http2FloodMitigationTest::prefillOutboundDownstreamQueue(uint32_t data_fram
...
+  // empty before forcing upstream socket to return EAGAIN. Envoy's upstream codec will send a few
+  // frames (i.e. SETTINGS and ACKs) after a new stream was established, and the test need to make
+  // sure these are flushed into the socket. To do so the test goes through the following steps:
+  // 1. send request headers, do not end stream
+  // 2. wait for headers to be received by the upstream and send response headers without ending
+  // stream
+  // 3. wait for client to receive response headers
+  // 4. send 1 byte of data from client
+  // 5. wait for 1 byte of data at the upstream. Receiving this DATA frame means that all other
+  // frames that Envoy sent before it are
```
Comment (asraa) [inline test/integration/http2_flood_integration_test.cc:290]: super nit: format comment with indent like below

File: test/integration/http2_flood_integration_test.cc [hunk 2]
```diff
@@ -274,6 +275,52 @@ void Http2FloodMitigationTest::prefillOutboundDownstreamQueue(uint32_t data_fram
   EXPECT_EQ(0, test_server_->counter("http2.outbound_flood")->value());
 }
 
+IntegrationStreamDecoderPtr
+Http2FloodMitigationTest::prefillOutboundUpstreamQueue(uint32_t frame_count) {
+  // This a bit complex exchange below is to ensure that the upstream outbound queue is
```
Comment (asraa) [inline test/integration/http2_flood_integration_test.cc:280]: super nit: remove bit/fix

**Review by asraa — CHANGES_REQUESTED** [review]
Thank you! LGTM 
/wait

File: test/integration/http2_flood_integration_test.cc (hunk 1 above)
Comment (yanavlasov) [reply test/integration/http2_flood_integration_test.cc]: done

**Review by asraa — APPROVED** [review]

**Pull request merged**